### PR TITLE
Add proxmox_node_info module

### LINKED ISSUE #7686: Proxmox: add module to retrieve node status / usage information
### Summary

I am developing a playbook to provision VMs to a Proxmox cluster, and I would like to be able to deploy each enw VM to the "least loaded" node of the cluster. This module will let me retrieve current load information - specifically CPU and memory usage - which I can then sort on to select the "least loaded" node.

### Issue Type

Feature Idea

### Component Name

proxmox_node_info

### Additional Information

```yaml (paste below)
- name: List existing nodes
  community.general.proxmox_node_info:
    api_host: proxmox1
    api_user: root@pam
    api_password: "{{ password | default(omit) }}"
    api_token_id: "{{ token_id | default(omit) }}"
    api_token_secret: "{{ token_secret | default(omit) }}"
  register: proxmox_nodes
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jwbernin**
Restarted PR due to erroneous update/p…

##### SUMMARY

Adds a module to retrieve basic information about Proxmox nodes. Design is based on existing proxmox_domain_info module

Fixes #7686 

##### ISSUE TYPE
- Feature Pull Request
- New Module/Plugin Pull Request

##### COMPONENT NAME
proxmox_node_info

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/proxmox_node_info.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      cpu:
+        description: Current CPU usage in fractional shares of this host's total available CPU.
+        returned: on success
+        type: float
+      disk:
+        description: Current local disk usage of this host.
+        returned: on success
+        type: int
+      id:
+        description: identity of the node
```
Comment (felixfontein) [inline plugins/modules/proxmox_node_info.py:55]:
```suggestion
        description: Identity of the node.
```

File: plugins/modules/proxmox_node_info.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      node:
+        description: Short hostname of this node.
+        returned: on success
+        type: str
+      ssl_fingerprint:
+        description: SSL fingerprint of the node certificate.
+        returned: on success
+        type: str
+      status:
+        description: Node status
```
Comment (felixfontein) [inline plugins/modules/proxmox_node_info.py:87]:
```suggestion
        description: Node status.
```

File: plugins/modules/proxmox_node_info.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+def main():
+    module_args = proxmox_auth_argument_spec()
+    node_info_args = proxmox_node_info_argument_spec()
+    module_args.update(node_info_args)
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        required_one_of=[('api_password', 'api_token_id')],
+        required_together=[('api_token_id', 'api_token_secret')],
+        supports_check_mode=True
```
Comment (felixfontein) [inline plugins/modules/proxmox_node_info.py:125]: Using a trailing comma makes it easier to add a new option without having to modify this line as well.
```suggestion
        supports_check_mode=True,
```

File: .github/BOTMETA.yml [hunk 4]
```diff
@@ -1038,6 +1038,8 @@ files:
   $modules/proxmox_template.py:
     ignore: skvidal
     maintainers: UnderGreen
+  $modules/proxmox_node_info.py:
```
Comment (felixfontein) [inline .github/BOTMETA.yml:1041]: Can you move this up a bit so it fits alphabetically between the other `proxmox_*` modules?

**Comment by felixfontein** [comment]
Please note that you need to add some form of tests. All new modules and plugins are required to have tests (unit tests and/or integration tests). See `tests/unit/plugins/modules/test_proxmox_*` and `tests/integration/targets/proxmox*` for examples of both.

**Comment by jwbernin** [comment]
Would it be okay to add tests to tests/integration/targets/proxmox/tasks/main.yml, or would it be preferable to create a tests/integration/proxmox_node_info directory?

**Comment by felixfontein** [comment]
tests/integration/targets/proxmox/tasks/main.yml should be fine, if you don't need some special setup that feels like it should better be its own test suite.

**Comment by felixfontein** [comment]
Looks good to me. If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@jwbernin thanks for your contribution!

**Pull request merged**